tests/gtests/test_reduction.cpp

```diff
@@ -49,6 +49,10 @@ class reduction_test_t
 
         SKIP_IF(unsupported_data_type(src_dt),
                 "Engine does not support this data type.");
+#if DNNL_GPU_VENDOR == DNNL_VENDOR_INTEL
+        SKIP_IF(get_test_engine().get_kind() != engine::kind::cpu,
+                "Engine does not support this primitive.");
+#endif
```

Comment:
The Intel GPU backend should have a bunch of implementation to pick from, for both OCL and SYCL runtimes.
So even though your previous PR made [those bugs](https://github.com/oneapi-src/oneDNN/pull/2243#issuecomment-2598761745) visible, I don't think rehiding these issues is the proper fix here.

Reply:
My bad, I assumed those tests had been disabled on purpose and I re-added the check. From a quick look, both `intel::ocl::atomic_reduction_t` & `intel::ocl::combined_reduction_t` implementations (which are the top 2 in the implementation list) can't handle 0-dimensional inputs correctly. `intel::ocl::ref_reduction_t` seems to work correctly.

Reply:
For the record, I've created #2469 which fixes these failures by handling the zero-dim cases "correctly".